handler/middleware.go

```diff
@@ -168,6 +169,15 @@ func (m *middleware) getOrCreateGuest(ctx context.Context) (*guest, error) {
 		if g, createErr := m.newGuest(ctx); createErr != nil {
 			return nil, createErr
 		} else {
+			runtime.SetFinalizer(g, func(g *guest) {
```

Comment:
I wonder if we truly need this given https://github.com/http-wasm/http-wasm-host-go/pull/86/files#diff-0b46d7cd26aa441d6e866b658afb5a9d61e9ff5f728d1524b62dcbf4b1309775R198

Reply:
I think it's OK. 

It makes it clear that if we call `runtime.Close()`, there is no need to close the guest again since `runtime.Close()` already does that.

Reply:
no. I do not think so.

the created `middleware` will not call `Close` in general. (and if it does , yes it prevent leak)

let's think the situation:

1.  traefik loaded and init a `middleware` 
2. user makes lots of request, the `Pool` size goes to 1000
3. user stop to make requests (in low traffic time)
4. the `Pool` begin GC (go runtime does this), but it can not free the memory properly.
5. now, the leak happens,  no user request, but the memory keeps

Reply:
@jcchavezs I a do some tests before make these PRs.

here is a simple do http app to simulate the GC: 

https://github.com/ttys3/traefik-wasm-plugin-memory-leak-demo/blob/main/go-sync-pool.go

```
env GODEBUG=gctrace=1 go run ./go-sync-pool.go

# in another term window
watch -n1 'ps_mem  | grep -E "traefik|go-sync-pool"'
```

so simulate the leak, just comment out https://github.com/ttys3/traefik-wasm-plugin-memory-leak-demo/blob/926a285b8486bbe607240037db7c1d97cf73cc09/go-sync-pool.go#L24 this line.

but in http-wasm-host-go, the leak is not the same. in here, it is golang GC can not handle the memory allocated by this go mod.

Reply:
Makes total sense.

Reply:
yes, it would be very nice, if we can close this PR and do not have memory leaks.

I have update the demo: https://github.com/ttys3/traefik-wasm-plugin-memory-leak-demo/blob/main/go-sync-pool.go

I try to use cgo to emulate the situation in http-wasm-host-go, but I failed.

even without `runtime.SetFinalizer` , my example will result in a good GC result.